src/main/java/reposense/model/Author.java

```diff
@@ -48,12 +59,25 @@ public Author(StandaloneAuthor sa) {
 
     public Author(Author another) {
         this.gitId = another.gitId;
+        this.emails = another.emails;
         this.displayName = another.gitId;
         this.authorAliases = another.authorAliases;
         this.ignoreGlobList = another.authorAliases;
         this.ignoreGlobMatcher = another.ignoreGlobMatcher;
     }
 
+    /**
+     * Checks that all the strings in the {@code emails} only contains commonly used email patterns.
+     * @throws IllegalArgumentException if any of the values do not meet the criteria.
+     */
+    private static void validateEmails(List<String> emails) throws IllegalArgumentException {
+        for (String email: emails) {
+            if (!email.matches(COMMON_EMAIL_REGEX)) {
+                throw new IllegalArgumentException(String.format(MESSAGE_UNCOMMON_EMAIL_PATTERN, email));
```

Comment:
What happens if one of the 2 emails provided is valid and the other is invalid? Do we still keep the valid email?

Reply:
This follows the current implementation of other validation. In the event that any of the emails are invalid, `IllegalArgumentException` will be thrown, prompt to the user that the config is misconfigured and will not be used (so nothing gets retained). 

Can open an issue for this to discuss if we should fix this in future.